### PR TITLE
merge from main into dev/4.3.x

### PR DESCRIPTION
**Pull request by yhilmare**
#### What this PR does / why we need it:
merge from main into dev/4.3.x
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by guowl3 — APPROVED** [review]
LGTM

**Review by PeachThinking — APPROVED** [review]
lgtm

**Review by LuckyPickleZZ — APPROVED** [review]
lgtm

**Review by krihy — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by ungreat — APPROVED** [review]
LGTM

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by Jane201510 — APPROVED** [review]

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 1]
```diff
@@ -151,7 +151,7 @@ com.oceanbase.odc.ErrorCodes.ConnectionTooManyPermissions=Too many permissions,
 com.oceanbase.odc.ErrorCodes.ExportExcelFileFailed=Failed to export the Excel file. It is recommended to export in the CSV format. Details: {0}.
 com.oceanbase.odc.ErrorCodes.OscSqlTypeInconsistent=The input SQL type {0} is inconsistent with the expected SQL type {1}. SQL: {2}.
 com.oceanbase.odc.ErrorCodes.NoUniqueKeyExists=The table {0} does not have a primary key index or non-null unique index. 
-com.oceanbase.odc.ErrorCodes.OscNotEnabled=Online schema change is not enabled in the current instance {0}. Please contact the administrator.
+com.oceanbase.odc.ErrorCodes.OscNotEnabled=Online schema change is not enabled in the current user {0}. Please contact the administrator.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties:154]: in the current user > for the current user

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 2]
```diff
@@ -137,7 +137,7 @@ com.oceanbase.odc.ErrorCodes.ObGetExecuteDetailFailed=Failed to fetch the execut
 com.oceanbase.odc.ErrorCodes.ObGetPlanExplainFailed=Failed to fetch the execution plan due to possible syntax error or invalid DDL statement. Details: {0}.
 com.oceanbase.odc.ErrorCodes.ObGetPlanExplainEmpty=Unable to fetch the execution plan due to possible syntax error, invalid DDL statement, or unsupported command. Details: {0}.
 com.oceanbase.odc.ErrorCodes.ObGetFullLinkTraceFailed=Failed to fetch information on end-to-end trace. Please check if the session variable 'ob_enable_show_trace' is set to 'ON'.
-com.oceanbase.odc.ErrorCodes.ObFullLinkTraceNotSupported=End-to-end trace is only available for OceanBase Database with versions higher than 4.0.
+com.oceanbase.odc.ErrorCodes.ObFullLinkTraceNotSupported=End-to-end trace is only available for OceanBase Database with versions equal to or greater than 4.2.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties:140]: greater > higher

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 3]
```diff
@@ -186,6 +186,8 @@ com.oceanbase.odc.ErrorCodes.ExecPLByAnonymousBlockUndefinedParameters=Unable to
 com.oceanbase.odc.ErrorCodes.ExecPLByAnonymousBlockErrorFormatParameters=The parameter passing method in line {0} does not conform to the definition. The parameter must be passed using the p => p format. Please do not modify the generated template.
 com.oceanbase.odc.ErrorCodes.ExecPLByAnonymousBlockDifferentParameters=The PL parameter {1} in line {0} is inconsistent with the assignment content {2}. Please do not delete or modify the generated template.
 com.oceanbase.odc.ErrorCodes.ExecPLByAnonymousBlockErrorCallStatement=The call to the PL object {0} cannot be found in the anonymous block. Please do not modify the generated anonymous block template.
+com.oceanbase.odc.ErrorCodes.PartitionPlanNoCreatePreviewSqlGenerated=No partition creation statement is generated. Possible reasons: The newly generated partition is a duplicate of the existing partition.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties:189]: The SQL statement for partition creation is not generated. Possible cause: The partition to be generated is a duplicate of the existing partition.

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 4]
```diff
@@ -186,6 +186,8 @@ com.oceanbase.odc.ErrorCodes.ExecPLByAnonymousBlockUndefinedParameters=Unable to
...
+com.oceanbase.odc.ErrorCodes.PartitionPlanNoDropPreviewSqlGenerated=No partition deletion statement is generated. Possible reason: No partition matches the deletion rule.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties:190]: The SQL statement for partition deletion is not generated. Possible cause: No partition matches the deletion rule.

**Review by JessieWuJiexi — APPROVED** [review]

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties (hunk 2 above)
Comment (yhilmare) [reply server/odc-core/src/main/resources/i18n/ErrorMessages.properties:140]: solve later

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties (hunk 3 above)
Comment (yhilmare) [reply server/odc-core/src/main/resources/i18n/ErrorMessages.properties:189]: solve later

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties (hunk 4 above)
Comment (yhilmare) [reply server/odc-core/src/main/resources/i18n/ErrorMessages.properties:190]: solve later

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**